packages/datadog-plugin-undici/test/index.spec.js

```diff
@@ -0,0 +1,528 @@
+'use strict'
+
+const getPort = require('get-port')
+const agent = require('../../dd-trace/test/plugins/agent')
+const tags = require('../../../ext/tags')
+const { expect } = require('chai')
+const { rawExpectedSchema } = require('./naming')
+const { DD_MAJOR } = require('../../../version')
+const { ERROR_MESSAGE, ERROR_TYPE, ERROR_STACK } = require('../../dd-trace/src/constants')
+
+const HTTP_REQUEST_HEADERS = tags.HTTP_REQUEST_HEADERS
+const HTTP_RESPONSE_HEADERS = tags.HTTP_RESPONSE_HEADERS
+
+const SERVICE_NAME = DD_MAJOR < 3 ? 'test-http-client' : 'test'
+
+describe('Plugin', () => {
+  let express
+  let fetch
+  let appListener
+
+  describe('undici-fetch', () => {
+    function server (app, port, listener) {
+      const server = require('http').createServer(app)
+      server.listen(port, 'localhost', listener)
+      return server
+    }
+
+    beforeEach(() => {
+      console.log('BE')
+      appListener = null
+    })
+
+    afterEach(() => {
+      console.log('AE')
+      if (appListener) {
+        appListener.close()
+      }
+      return agent.close({ ritmReset: false })
+    })
+
+    describe('without configuration', () => {
+      beforeEach(async () => {
+      console.log('BEE')
+        return await agent.load('undici', {
+          service: 'test'
+        })
+          .then(() => {
+            console.log('BEEX')
+            express = require('express')
+            fetch = require('undici')
```

Comment:
Should probably change these to use `withVersions(...)` so we get version checking and don't need to add undici as a dev dependency.

Reply:
This is done.